spec/requests/api/v1/topics/index_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+# Test suite for GET /api/v1/topics
+describe 'GET /api/v1/topics', type: :request do
+  let(:user) { create(:user) }
+  let!(:topics) { create_list(:topic, 5) }
+
+  before(:each) do
+    user.confirm
+  end
+
+  context 'when the request is valid' do
+    before { get '/api/v1/topics', headers: auth_headers(user) }
+
+    it 'returns all topics' do
+      expect(json).not_to be_empty
+      expect(json['topics'].size).to eq(5)
```

Comment:
Commonly, it's better to avoid using magic numbers, in this case it's easy to match this number 5 with the amount of topics, but sometimes it isn't.
`expect(json['topics'].size).to eq(topics.size)`